src/driver/src/DefaultDriver.cpp

```diff
@@ -241,9 +100,13 @@ DefaultDriver::OutcomeCode doRunLogic(const DefaultDriver::ProgramOptions& optio
         phaseConfigSource = fs::path(options.workloadSource).parent_path();
     }
 
-    YamlParameters params;
-    auto config = loadConfig(options.workloadSource, options.workloadSourceType);
-    auto yaml = recursiveParse(config, params, phaseConfigSource);
+    v1::WorkloadParser parser{phaseConfigSource};
+    auto yaml = parser.parse(options.workloadSource, options.workloadSourceType);
+
+    if (options.isSmokeTest) {
+        yaml = v1::SmokeTestConverter::convert(yaml);
+    }
```

Comment:
hm...seems like the parser should be able to know how to do all the parsing - e.g. pass in `options.workloadSource, options.workloadSourceType, options.isSmokeTest` and have it return the fully-parsed thing?

Reply:
Yeah good idea.